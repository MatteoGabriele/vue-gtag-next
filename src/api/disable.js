import { allProperties } from "@/state";

export default (value = true) => {
  allProperties.value.forEach((property) => {
    window[`ga-disable-${property.id}`] = value;
  });
};
