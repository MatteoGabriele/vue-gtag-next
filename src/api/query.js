import { useState } from "@/state";

export default (...args) => {
  if (typeof document === "undefined" || typeof window === "undefined") {
    return;
  }

  const { globalObjectName, useDebugger } = useState();

  if (useDebugger.value) {
    console.warn("[vue-gtag] Debugger:", args);
  }

  window[globalObjectName.value](...args);
};
