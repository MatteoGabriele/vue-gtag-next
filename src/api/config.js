import { allProperties } from "@/state";
import query from "@/api/query";

export default (...args) => {
  allProperties.value.forEach((property) => {
    query("config", property.id, ...args);
  });
};
