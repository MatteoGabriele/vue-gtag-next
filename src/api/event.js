import query from "@/api/query";
import { allProperties } from "@/state";

export default (eventName, eventParams = {}) => {
  const params = { ...eventParams };

  if (!params.send_to && allProperties.value.length > 1) {
    params.send_to = allProperties.value.map((property) => property.id);
  }

  query("event", eventName, params);
};
