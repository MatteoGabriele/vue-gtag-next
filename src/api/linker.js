import config from "@/api/config";

export default (...args) => {
  config("linker", ...args);
};
