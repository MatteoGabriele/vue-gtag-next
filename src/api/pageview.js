import event from "@/api/event";

export default (value) => {
  let params = {};

  if (typeof value === "string") {
    params = {
      page_path: value,
      page_location: window.location.href,
    };
  } else {
    params = value;
  }

  if (typeof params.send_page_view === "undefined") {
    params.send_page_view = true;
  }

  event("page_view", params);
};
