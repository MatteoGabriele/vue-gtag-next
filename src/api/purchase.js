import event from "@/api/event";

export default (...args) => {
  event("purchase", ...args);
};
