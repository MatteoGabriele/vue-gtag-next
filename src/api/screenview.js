import { useState } from "@/state";
import event from "@/api/event";

export default (...args) => {
  const { appName } = useState();
  const [arg] = args;
  let params = {};

  if (typeof arg === "string") {
    params = {
      screen_name: arg,
    };
  } else {
    params = arg;
  }

  if (params.app_name == null) {
    params.app_name = appName;
  }

  if (params.send_page_view == null) {
    params.send_page_view = true;
  }

  event("screen_view", params);
};
