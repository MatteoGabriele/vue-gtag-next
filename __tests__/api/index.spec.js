import * as api from "@/api";

describe("api", () => {
  it("should render the api object", () => {
    expect(Object.keys(api)).toMatchSnapshot();
  });
});
