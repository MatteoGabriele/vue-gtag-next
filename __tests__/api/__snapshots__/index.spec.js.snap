// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api should render the api object 1`] = `
Array [
  "config",
  "customMap",
  "disable",
  "event",
  "exception",
  "linker",
  "pageview",
  "purchase",
  "query",
  "refund",
  "screenview",
  "set",
  "time",
]
`;
